runners/spark/src/test/java/org/apache/beam/runners/spark/ProvidedSparkContextTest.java

```diff
@@ -64,21 +47,25 @@ private SparkContextOptions getSparkContextOptions(JavaSparkContext jsc) {
     @Test
     public void testWithProvidedContext() throws Exception {
         JavaSparkContext jsc = new JavaSparkContext("local[*]", "Existing_Context");
+        testWithValidProvidedContext(jsc);
+        if (!jsc.sc().isStopped()) {
+            jsc.stop();
+        }
+    }
 
-        SparkContextOptions options = getSparkContextOptions(jsc);
-
-        Pipeline p = Pipeline.create(options);
-        PCollection<String> inputWords = p.apply(Create.of(WORDS).withCoder(StringUtf8Coder
-                .of()));
-        PCollection<String> output = inputWords.apply(new WordCount.CountWords())
-                .apply(MapElements.via(new WordCount.FormatAsTextFn()));
-
-        PAssert.that(output).containsInAnyOrder(EXPECTED_COUNT_SET);
-
-        // Run test from pipeline
-        p.run().waitUntilFinish();
-
-        jsc.stop();
+    /**
+     * Run two pipelines reusing the context.
+     * @throws Exception
+     */
+    @Test
+    public void testReuseProvidedContext() throws Exception {
```

Comment:
This could be avoided by asserting the context stays alive after executing a provided context pipeline (mentioned above).